pkg/build/openbsd.go

```diff
@@ -15,25 +15,25 @@ import (
 
 type openbsd struct{}
 
-func (ctx openbsd) build(params *Params) error {
+func (ctx openbsd) build(params Params) (compilerID string, err error) {
```

Comment:
Per https://tour.golang.org/basics/7
```
Naked return statements should be used only in short functions...
They can harm readability in longer functions.
```
I suspect we are stretching the definition of "short" here.

Reply:
I agree on that. Maybe I'm missing something, but for me this decision seemed like the choice of the lesser of the evils. It would be great to hear your perspective on what would better align with the Golang approaches.

1) Explicitly return a pair of values from everywhere. There are already tons of duplicating returns inside pkg/build/{OS}.go files, and this would make them all more complex. Extracting code into other methods like it is done in linux.go could possibly make this less of a problem, though.
2) Use named return statements. That was already commented by you above.
3) Method overloading (so that only the OSes, which can provide the data, would bother actually returning the value) - this is not supported by Go.
4) Complicate the `builder` interface by introducing two separate `build` methods with different names (one that only returns an error and one that also returns a compiler ID) - did not seem to be worth it.
5) Add an "output" parameter, like `build(params Params, compilerID *string)`? Does not seem to be an idiomatic Go approach.

Reply:
Here, due to `compilerID` being completely unused, I see an option of keeping the original `build` under a private name, and then having the new `build` one delegate to it. What do you think?

Reply:
@blackgnezdo we will go this the current version. We've had some discussions privately, and it seems there is no perfect solution. We don't use naked returns anymore. The diff is quite large, but it now allows any OS implementation to easily return any of the ImageResult fields and it allows easy addition of new ImageResult fields.